engine/table/src/main/java/io/deephaven/engine/table/impl/SourcePartitionedTable.java

```diff
@@ -248,45 +248,46 @@ private RowSet processAdditions(final TableLocationSubscriptionBuffer.LocationUp
         }
 
         private RowSet processRemovals(final TableLocationSubscriptionBuffer.LocationUpdate locationUpdate) {
-            final Set<ImmutableTableLocationKey> relevantRemovedLocations = locationUpdate.getPendingRemovedLocations()
-                    .stream()
-                    .map(TableLocation::getKey)
-                    .filter(locationKeyMatcher)
-                    .collect(Collectors.toSet());
+            final Set<ImmutableTableLocationKey> relevantRemovedLocations =
+                    locationUpdate.getPendingRemovedLocationKeys()
+                            .stream()
+                            .filter(locationKeyMatcher)
+                            .collect(Collectors.toSet());
 
             if (relevantRemovedLocations.isEmpty()) {
                 return RowSetFactory.empty();
             }
 
+            // At the end of the cycle we need to make sure we unmanage any removed constituents.
+            this.removedConstituents = new ArrayList<>(relevantRemovedLocations.size());
+            this.removedLocationsComitter = new UpdateCommitter<>(
+                    this,
+                    result.getUpdateGraph(),
+                    ignored -> {
+                        removedConstituents.forEach(result::unmanage);
+                        removedConstituents = null;
+                    });
+            this.removedLocationsComitter.maybeActivate();
```

Comment:
I would rather have you activate this once you've finished populating `removedConstituents` if its non-empty.